### PR TITLE
Auth 2

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: src/ts/Login/ChangeSuccess.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    />,
+  ]
+
+  return (
+    <Dialog
+      open={open}
+      contentStyle={{maxWidth: 650}}
+      title="Password changed successfully"
+      actions={actions}
+    >
```
Comment (michalmikolajczyk) [inline src/ts/Login/ChangeSuccess.tsx]: inline

File: src/ts/Login/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    this.submit = this.submit.bind(this)
+  }
+
+  submit(user) {
+    this.setState({canSubmit: false})
+    login(user)
+    .then(res => {
+      if (res['err']) {
+        this.setState({openDialog: true})
+      } else {
```
Comment (michalmikolajczyk) [inline src/ts/Login/LoginForm.tsx]: `else` unneeded with `return` in `if`

File: src/ts/Offers/index.tsx [hunk 3]
```diff
@@ -96,12 +101,36 @@ export default class Offers extends React.Component<any, any> {
...
 
   componentDidMount() {
-    this.loadMore();
+    this.loadMore()
+    if (this.props.params.token) {
+      this.verifyAccount()
+    }
+  }
+
+  verifyAccount() {
```
Comment (michalmikolajczyk) [inline src/ts/Offers/index.tsx]: this function does not belong in the offers component. It should probably be higher up in the hierarchy

File: src/ts/Offers/index.tsx [hunk 4]
```diff
@@ -96,12 +101,36 @@ export default class Offers extends React.Component<any, any> {
...
+    let token = this.props.params.token
+    verify({token})
+    .then(res => {
```
Comment (michalmikolajczyk) [inline src/ts/Offers/index.tsx]: chain indent please

File: src/ts/Signin/SigninDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        label="Login"
+        primary={true}
+      />
+    </div>,
+    <FlatButton
+      label="Cancel"
+      primary={true}
+      keyboardFocused={true}
+      onTouchTap={close}
+    />,
```
Comment (michalmikolajczyk) [inline src/ts/Signin/SigninDialog.tsx]: those trailing commas can sometimes cause errors

File: src/ts/Signin/index.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import Container from '../Container'
+import SigninForm from './SigninForm'
+
+export default function Signin(props) {
+  return (
+    <Container className="cont">
+      <div className="form">
+        <h1 className="title">Create a new account</h1>
+        <div className="title-sub">
+          Already have an account?&#160;
```
Comment (michalmikolajczyk) [inline src/ts/Signin/index.tsx]: whats the html character here?

File: src/ts/Login/LoginForm.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        let token = res['token']
+        let user = res['user']
+        localStorage.setItem('jwt', token)
```
Comment (jonzee) [inline src/ts/Login/LoginForm.tsx]: seems ok http://caniuse.com/#search=localStorage

File: src/ts/Login/LoginForm.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        localStorage.setItem('user', JSON.stringify(user))
```
Comment (jonzee) [inline src/ts/Login/LoginForm.tsx]: ok, added `localStorage` polyfill

File: src/ts/Login/LoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  render() {
+    return (
+      <Form
+        onValid={() => this.setState({canSubmit: true})}
+        onInvalid={() => this.setState({canSubmit: false})}
+        onSubmit={this.submit}
+      >
+        <FormsyText
+          name="email"
+          value="viking.garage.app@gmail.com"
```
Comment (jonzee) [inline src/ts/Login/LoginForm.tsx]: It was for fast smoke test with `API`

File: src/ts/Menu/index.tsx [hunk 10]
```diff
@@ -7,8 +8,14 @@ import {
...
-  const user = true
+  function logout() {
+    localStorage.removeItem('jwt')
+    localStorage.removeItem('user')
+    props.update()
+    browserHistory.push('/')
+  }
 
+  const user = localStorage.getItem('user')
   const userImage = "https://scontent-waw1-1.cdninstagram.com/t51.2885-15/e35/15877344_344661349253204_5404343355553349632_n.jpg"
```
Comment (jonzee) [inline src/ts/Menu/index.tsx]: I hope so, it's your photo from insta 😀

File: src/ts/Login/Reset.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
```
Comment (jonzee) [inline src/ts/Login/Reset.tsx]: Here we have some forms/api request so we need to keep state

File: src/ts/Signin/index.tsx (hunk 6 above)
Comment (jonzee) [inline src/ts/Signin/index.tsx]: space char, I changed it to no-breaking space `&nbsp;` http://andrewhfarmer.com/how-whitespace-works-in-jsx/

**Pull request merged**